fastcrawler/engine/contracts.py

```diff
@@ -18,15 +20,17 @@ class SetCookieParam(pydantic.BaseModel):
     sameSite: str | Literal["Lax", "None", "Strict"] = ""
 
 
-class ProxySetting(pydantic.BaseModel):
-    protocol: str = "http://"
+@dataclass
+class ProxySetting:
     server: str
     port: int
     username: str | None = None
     password: str | None = None
+    protocol: str = "http://"
```

Comment:
this should be an enum

Reply:
will be done in another pr. 
btw, what are the possibilities? I should research.